### DISCUSSION TITLE
Using PARANOID leak detection level at production environment

### DISCUSSION THREAD

**Discussion by MintNat** [netty/netty#14123] [Q&A]
Hello.
There are couple memory leaks, which we managed to find at testing environment for complex system, with simply bumping up the detection leak level to paranoid, so I'm wondering is this a production feature? I'm having concerns about performance. Are there any recommendations with this regard or someone with relatable experience?

**Comment by slandelle** [comment]
Huge performance drop for high throughput system. If your throughput is 10 rps, you obviously don't care.

**Reply by MintNat** [reply]
As I thought. Thank you. How about Advanced level then?

**Reply by He-Pin** [reply]
You can turn it on in staging and find issues with stressing test, but turn it off in production

**Comment by normanmaurer** [comment]
As others said... you should never use anything else then `SIMPLE` in prod.
